geometry.go

```diff
@@ -302,36 +303,51 @@ func (m Matrix) Scaled(around Vec, scale float64) Matrix {
 
 // Rotated rotates everything around a given point by the given angle in radians.
 func (m Matrix) Rotated(around Vec, angle float64) Matrix {
-	m3 := mgl64.Mat3(m)
-	m3 = mgl64.Translate2D(around.Scaled(-1).XY()).Mul3(m3)
-	m3 = mgl64.Rotate3DZ(angle).Mul3(m3)
-	m3 = mgl64.Translate2D(around.XY()).Mul3(m3)
-	return Matrix(m3)
+	sint, cost := math.Sin(angle), math.Cos(angle)
+	m[2], m[5] = m[2]-around.X, m[2]-around.Y
+	m = m.Chained(Matrix{cost, -sint, 0, sint, cost, 0})
+	m[2], m[5] = m[2]+around.X, m[2]+around.Y
+	return m
 }
 
 // Chained adds another Matrix to this one. All tranformations by the next Matrix will be applied
 // after the transformations of this Matrix.
 func (m Matrix) Chained(next Matrix) Matrix {
-	m3 := mgl64.Mat3(m)
-	m3 = mgl64.Mat3(next).Mul3(m3)
-	return Matrix(m3)
+	return Matrix{
+		m[0] * next[0] + m[1] * next[3],
+		m[0] * next[1] + m[1] * next[4],
+		m[0] * next[2] + m[1] * next[5] + m[2],
+		m[3] * next[0] + m[4] * next[3],
+		m[3] * next[1] + m[4] * next[4],
+		m[3] * next[2] + m[4] * next[5] + m[5],
+	}
 }
 
 // Project applies all transformations added to the Matrix to a vector u and returns the result.
 //
 // Time complexity is O(1).
 func (m Matrix) Project(u Vec) Vec {
-	m3 := mgl64.Mat3(m)
-	proj := m3.Mul3x1(mgl64.Vec3{u.X, u.Y, 1})
-	return V(proj.X(), proj.Y())
+	return V(m[0] * u.X + m[1] * u.Y + m[2], m[3] * u.X + m[4] * u.Y + m[5])
```

Comment:
For consistency (and maybe performance), I'd use `Vec{...}` literal here instead of calling `V(...)` constructor.